packages/components/src/components/radio/radio.component.ts

```diff
@@ -0,0 +1,233 @@
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { NameMixin } from '../../utils/mixins/NameMixin';
+import { ValueMixin } from '../../utils/mixins/ValueMixin';
+import styles from './radio.styles';
+import FormfieldWrapper from '../formfieldwrapper/formfieldwrapper.component';
+
+/**
+ * Radio allow users to select single options from a list or turn an item/feature on or off.
+ * These are often used in forms, settings, and selection in lists.
+ *
+ * A radio component contains an optional label, optional info icon and an optional helper text.
+ * @dependency mdc-formfieldwrapper
+ *
+ * @tagname mdc-radio
+ *
+ * @cssproperty --mdc-radio-inner-circle-size - size of the inner circle
+ * @cssproperty --mdc-radio-text-disabled-color - color of the label when disabled
+ * @cssproperty --mdc-radio-disabled-border-color - color of the radio button border when disabled
+ * @cssproperty --mdc-radio-normal-border-color - color of the radio button border when normal
+ * @cssproperty --mdc-radio-inner-circle-normal-background - background color of the inner circle when normal
+ * @cssproperty --mdc-radio-inner-circle-disabled-background - background color of the inner circle when disabled
+ * @cssproperty --mdc-radio-control-inactive-color - color of the radio button when inactive
+ * @cssproperty --mdc-radio-control-inactive-hover - color of the radio button when inactive and hovered
+ * @cssproperty --mdc-radio-control-inactive-pressed-color - color of the radio button when inactive and pressed
+ * @cssproperty --mdc-radio-control-inactive-disabled-background - background color of the radio button when
+ *  inactive and disabled
+ * @cssproperty --mdc-radio-control-active-color - color of the radio button when active
+ * @cssproperty --mdc-radio-control-active-hover-color - color of the radio button when active and hovered
+ * @cssproperty --mdc-radio-control-active-pressed-color - color of the radio button when active and pressed
+ * @cssproperty --mdc-radio-control-active-disabled-background - background color of the radio button
+ *  when active and disabled
+ *
+ */
+
+class Radio extends NameMixin(ValueMixin(FormfieldWrapper)) {
+  /**
+  * Determines whether the radio is selected or unselected.
+  *
+  * @default false
+  */
+  @property({ type: Boolean, reflect: true }) checked = false;
+
+  /**
+  * Determines whether the radio is read-only.
+  *
+  * @default false
+  */
+  @property({ type: Boolean, reflect: true }) readonly = false;
+
+  /**
+   * Specifies the aria-label attribute for accessibility purposes.
+   */
+  @property({ type: String, reflect: true, attribute: 'data-aria-label' }) dataAriaLabel: string = '';
+
+  /** @internal */
+  private internals: ElementInternals;
+
+  /** @internal */
+  static formAssociated = true;
+
+  /** @internal */
+  get form(): HTMLFormElement | null {
+    return this.internals.form;
+  }
+
+  constructor() {
+    super();
+    this.internals = this.attachInternals();
```

Comment:
```suggestion
    super();
    /** @internal */
    this.internals = this.attachInternals();
```

Without this, it would show internals on storybook attributes.